src/main.cpp

```diff
@@ -1234,8 +1234,11 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
             return state.DoS(0, false, REJECT_NONSTANDARD, "too-long-mempool-chain", false);
         }
 
-        unsigned int forkVerifyFlags = ((Opt().UAHFTime() != 0) && (chainActive.Tip()->GetMedianTimePast() >= Opt().UAHFTime()) ?
+        const int64_t mtpChainTip = chainActive.Tip()->GetMedianTimePast();
+        unsigned int forkVerifyFlags = ((Opt().UAHFTime() != 0) && (mtpChainTip >= Opt().UAHFTime()) ?
```

Comment:
This could use `IsUAHFActive` from utilfork instead

Reply:
Sounds great